Merge request

sinattieng opened the pull request.

jamesward commented on .project:
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
This file should be gitignored

sinattieng replied:
Done.

jamesward commented:
With WebJars we pull the contents from the upstream since it is a much more maintainable approach.  Is it possible to rework this PR to do that?

sinattieng commented:
> With WebJars we pull the contents from the upstream since it is a much more maintainable approach. Is it possible to rework this PR to do that?

Sorry, that's my first webjars project, so in order to understand what you mean, and the correct process : you are talking about the upstream.url on pom, am i right? because, I have commented it for this error (mvn clean install on project):
<pre>
      [get] Getting: https://unpkg.com/browse/highlightjs-badge@0.1.7/highlightjs-badge.js
      [get] To: /mnt/ws/parer/highlightjs-badge/target/classes/META-INF/resources/webjars/highlightjs-badge/0.1.7-SNAPSHOT/highlightjs-badge.js
      [get] Error opening connection java.io.FileNotFoundException: https://unpkg.com/browse/highlightjs-badge@0.1.7/highlightjs-badge.js
      [get] Error opening connection java.io.FileNotFoundException: https://unpkg.com/browse/highlightjs-badge@0.1.7/highlightjs-badge.js
      [get] Error opening connection java.io.FileNotFoundException: https://unpkg.com/browse/highlightjs-badge@0.1.7/highlightjs-badge.js
      [get] Can't get https://unpkg.com/browse/highlightjs-badge@0.1.7/highlightjs-badge.js to /mnt/ws/parer/highlightjs-badge/target/classes/META-INF/resources/webjars/highlightjs-badge/0.1.7-SNAPSHOT/highlightjs-badge.js
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.068 s
[INFO] Finished at: 2020-03-04T18:27:41+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.8:run (default) on project highlightjs-badge: An Ant BuildException has occured: Can't get https://unpkg.com/browse/highlightjs-badge@0.1.7/highlightjs-badge.js to /mnt/ws/parer/highlightjs-badge/target/classes/META-INF/resources/webjars/highlightjs-badge/0.1.7-SNAPSHOT/highlightjs-badge.js
[ERROR] around Ant part ...<get src="https://unpkg.com/browse/highlightjs-badge@${version.unsnapshot}/highlightjs-badge.js" dest="/mnt/ws/parer/highlightjs-badge/target/classes/META-INF/resources/webjars/highlightjs-badge/0.1.7-SNAPSHOT"/>... @ 5:215 in /mnt/ws/parer/highlightjs-badge/target/antrun/build-main.xml
</pre>
Unfortunatelly, that link does not exist (and that's correct i think), what's wrong ? 

Stefano

sinattieng commented:
@jamesward 

I have fix the problem of upstream, like this : https://github.com/webjars/nprogress/blob/master/pom.xml, please verify if it is correct.

Stefano

jamesward commented:
The problem is that there doesn't seem to be a `0.1.7` release on NPM:
https://www.npmjs.com/package/highlightjs-badge

I used unpkg.com because it's an easy way to get NPM package contents and that seems to be the only place where the upstream project has tagged releases.

sinattieng commented:
> The problem is that there doesn't seem to be a `0.1.7` release on NPM:
> https://www.npmjs.com/package/highlightjs-badge
> 
> I used unpkg.com because it's an easy way to get NPM package contents and that seems to be the only place where the upstream project has tagged releases.

Ok, did you see my last commit? it should be resolve the issue.

Stefano

jamesward commented on pom.xml:
```diff
@@ -18,8 +18,9 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <upstream.url>https://unpkg.com/browse/highlightjs-badge@${version.unsnapshot}/</upstream.url>
-        <destDir>${project.build.outputDirectory}/META-INF/resources/webjars/${project.artifactId}/${project.version}</destDir>
+        <upstream.version>0.1.6</upstream.version>
```
You should be able to omit this and instead use `${version.unrevise}`

jamesward commented:
You should be able to remove `src/main/resources` if you are using the upstream downloads.

jamesward commented:
I'm missing what you are trying to update.  I used the latest release `0.1.6`.  Do you just need additional files from that release?

sinattieng commented:
> I'm missing what you are trying to update. I used the latest release `0.1.6`. Do you just need additional files from that release?

Exactly, I need to push some modification to the original js, so it means new release (0.1.7) : 
- i have found some bugs on the original js 
- i have added new functionalities which are needed 

So i have modified both js and min.js

jamesward commented:
Ah, I see.  Usually those changes go into the upstream project, then they do a new release, which we can then pull into the WebJar.  WebJars are just redistributions of external projects.  Is the upstream project still maintained?

sinattieng commented:
> Ah, I see. Usually those changes go into the upstream project, then they do a new release, which we can then pull into the WebJar. WebJars are just redistributions of external projects. Is the upstream project still maintained?

Really I don't know, that's the page of the original project https://weblog.west-wind.com/posts/2019/Dec/30/A-HighlightJs-Copy-Code-Badge-Component, it seems maintend but i need my modification in order to have the right "version" of the plugin (in the other case i should upload it on my own repository ... but i prefere having it on maven central).

jamesward commented:
If your change is generally useful then it should go into: https://github.com/RickStrahl/highlightjs-badge
If it's really just for you, then you should publish a fork.  If you publish a fork in NPM, then you could make your fork work with NPM WebJars which would enable you to publish the WebJar on your own to Maven Central.

sinattieng commented:
> If your change is generally useful then it should go into: https://github.com/RickStrahl/highlightjs-badge
> If it's really just for you, then you should publish a fork. If you publish a fork in NPM, then you could make your fork work with NPM WebJars which would enable you to publish the WebJar on your own to Maven Central.

Ok, I have forked and fixed the original project https://github.com/sinattieng/highlightjs-badge-1.git (unfortunatelly i have the postfix -1 becuase the other fork webjar project), so in order to publish my NPM WebJars I have tried with https://www.webjars.org/ deploy webjars NPM by the git repo url but i did something wrong ....

jamesward commented:
I'd recommend renaming the WebJar fork to `webjar-highlightjs-badge` so you can rename the other one to not have the `-1`.

jamesward commented:
BTW, when I tried the NPM deploy of the upstream it failed due to a missing field in the `package.json` but you should be able to fix that in your fork.

sinattieng commented:
Something goes wrong, I have published my npm, during the deploy i got this error:
<code>
{"status":"Validation Failed","messages":"[Failed to close repository: orgwebjars-63627., Dropping existing partial staging repository.]"}
</code>
after the maven central sync process.

https://github.com/sinattieng/highlightjs-badge.git

jamesward commented:
It looks like there aren't any tagged releases in your fork.  Not sure how it let you do a deploy...
